### PR TITLE
Hidden Indicators for rich data view

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
* Hidden indicators implementation for rich data view 
* Hide and show indicators, category and subcategory when visibility changed from my view
* Update rich data view on popstate events for hidden indicators
* Hide indicators on load if hidden indicators present in share url
* Hide nav option in rich data view when category is hidden

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-648

## How is it tested automatically?
* Check if indicators are hidden
* Check if HTML blocks should be hidden
* Check if subcategory and category are hidden if there are no indicators
* Check if rich data view scroll menu gets updated when category visibility is changed

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally
- [ ] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out
- [ ] commit messages are meaningful

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/profile/subcategory.js [hunk 1]
```diff
@@ -167,13 +175,12 @@ export class Subcategory extends Component {
                     $(wrapper).append(indicatorContainer);
                     let metadata = indicator.metadata;
                     if (indicator.content_type === ContentBlock.BLOCK_TYPES.Indicator) {
-                        block = this.addIndicatorBlock(indicatorContainer, indicator, title, isLast, addLockButton, restrictValues, defaultFilters);
+                        block = this.addIndicatorBlock(indicatorContainer, indicator, title, isLast, addLockButton, restrictValues, defaultFilters, hiddenIndicators);
                     } else if (indicator.content_type === ContentBlock.BLOCK_TYPES.HTMLBlock) {
-                        block = this.addHTMLBlock(indicatorContainer, indicator, title, isLast);
+                        block = this.addHTMLBlock(indicatorContainer, indicator, title, isLast, true, {}, [], hiddenIndicators);
```
Comment (michaelglenister) [inline src/js/profile/subcategory.js:180]: It looks like addHTMLBlock only expects four parameters

Comment (goyal1092) [reply src/js/profile/subcategory.js]: Thanks @michaelglenister I have updated code  👍

**Review by michaelglenister — APPROVED** [review]
Approved

**Review by michaelglenister — APPROVED** [review]
Tests look good
